astropy/units/utils.py

```diff
@@ -202,15 +202,8 @@ def validate_power(p, support_tuples=False):
     p : float, int, Rational, Fraction
         Power to be converted
     """
-    if isinstance(p, (numbers.Rational, Fraction)):
```

Comment:
`isinstance` is slow, especially on an ABC like `Rational`. The duck-typing makes more sense here, and is much faster.